crates/miden-objects/src/account/component/metadata/storage/type_registry.rs

```diff
@@ -0,0 +1,615 @@
+        SchemaTypeIdentifier::new("felt").expect("type is well formed")
+    }
+
+    /// Returns the schema type identifier for the native [`Word`] type.
+    pub fn native_word() -> SchemaTypeIdentifier {
+        SchemaTypeIdentifier::new("word").expect("type is well formed")
+    }
+
+    /// Returns the schema type identifier for storage map slots.
+    pub fn storage_map() -> SchemaTypeIdentifier {
+        SchemaTypeIdentifier::new("map").expect("type is well formed")
+    }
+
+    /// Returns the schema type identifier for the `void` type.
+    ///
+    /// The `void` type always parses to `0` and is intended to model reserved or padding felts.
+    pub fn void() -> SchemaTypeIdentifier {
+        SchemaTypeIdentifier::new("void").expect("type is well formed")
+    }
+
+    /// Returns a reference to the inner string.
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Display for SchemaTypeIdentifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+impl Serializable for SchemaTypeIdentifier {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write(self.0.clone())
+    }
+}
+
+impl Deserializable for SchemaTypeIdentifier {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let id: String = source.read()?;
+
+        SchemaTypeIdentifier::new(id)
+            .map_err(|err| DeserializationError::InvalidValue(err.to_string()))
+    }
+}
+
+// INIT VALUE REQUIREMENT
+// ================================================================================================
+
+/// Describes the expected type and additional metadata for an init-provided storage value.
+///
+/// An `InitValueRequirement` specifies the expected type identifier for a required init value as
+/// well as an optional description. This information is used to validate and provide context for
+/// instantiation-time values.
+#[derive(Debug)]
+pub struct InitValueRequirement {
+    /// The expected type identifier.
+    pub r#type: SchemaTypeIdentifier,
+    /// An optional description providing additional context.
+    pub description: Option<String>,
+}
+
+// SCHEMA TYPE TRAITS
+// ================================================================================================
+
+/// Trait for converting a string into a single `Felt`.
+pub trait FeltType: Send + Sync {
+    /// Returns the type identifier.
+    fn type_name() -> SchemaTypeIdentifier
+    where
+        Self: Sized;
+
+    /// Parses the input string into a `Felt`.
+    fn parse_felt(input: &str) -> Result<Felt, SchemaTypeError>
+    where
+        Self: Sized;
+
+    /// Displays a `Felt` in a canonical string representation for this type.
+    fn display_felt(value: Felt) -> Result<String, SchemaTypeError>
+    where
+        Self: Sized;
+}
+
+/// Trait for converting a string into a single `Word`.
+pub trait WordType: alloc::fmt::Debug + Send + Sync {
```

Comment:
Nit: Why does this trait require `Debug` but `FeltType` does not? Should they both require it?

Reply:
I've removed this. I tried different approaches to enable displaying types (for re-serialization, but also for enabling displaying values in places like the explorer, etc.) and I think this was a leftover from one of those attempts